other/word_search.py

```diff
@@ -0,0 +1,395 @@
+        >>> ws.insert_south_east("cat", [0], [0])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [['c', None, None],
+        [None, 'a', None],
+        [None, None, 't']]
+        >>> ws.insert_south_east("at", [1, 0], [2, 1, 0])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [['c', None, None],
+        ['a', 'a', None],
+        [None, 't', 't']]
+        """
+        word_length = len(word)
+        # Attempt to insert the word into each row and when successful, exit
+        for row in rows:
+            # Check if there is space for the word above the row
+            if word_length + row > self.height:
+                continue
+
+            # Attempt to insert the word into each column
+            for col in cols:
+                # Check if there is space to the right of the word as well as above
+                if word_length + col > self.width:
+                    continue
+
+                # Check if there are existing letters
+                # to the right of the column that will be overwritten
+                letters_diagonal_left = [
+                    self.board[row + i][col + i] for i in range(word_length)
+                ]
+                if all(letter is None for letter in letters_diagonal_left):
+                    # Successful, insert the word north
+                    for i in range(word_length):
+                        self.board[row + i][col + i] = word[i]
+                    return
+
+    def insert_south(self, word: str, rows: list[int], cols: list[int]) -> None:
+        """
+        >>> ws = WordSearch(WORDS, 3, 3)
+        >>> ws.insert_south("cat", [0], [0])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [['c', None, None],
+        ['a', None, None],
+        ['t', None, None]]
+        >>> ws.insert_south("at", [2, 1, 0], [0, 1, 2])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [['c', None, None],
+        ['a', 'a', None],
+        ['t', 't', None]]
+        """
+        word_length = len(word)
+        # Attempt to insert the word into each row and when successful, exit
+        for row in rows:
+            # Check if there is space below the row to fit in the word
+            if word_length + row > self.height:
+                continue
+
+            # Attempt to insert the word into each column
+            for col in cols:
+                # Only check to be made here is if there are existing letters
+                # below the column that will be overwritten
+                letters_below = [self.board[row + i][col] for i in range(word_length)]
+                if all(letter is None for letter in letters_below):
+                    # Successful, insert the word south
+                    for i in range(word_length):
+                        self.board[row + i][col] = word[i]
+                    return
+
+    def insert_south_west(self, word: str, rows: list[int], cols: list[int]) -> None:
```

Comment:
```suggestion
    def insert_southwest(self, word: str, rows: list[int], cols: list[int]) -> None:
```
Nitpick: `south_west` -> `southwest`